File: test/corpus/julia/duplicates.jl
# An exact clone pair: two functions identical up to renamed identifiers, each well
# above the clone size floor, reported as a `:duplicate` cluster.

# dendro-expect: duplicate
function total_price(items, tax)
    subtotal = 0
    for item in items
        subtotal = subtotal + item
    end
    grand = subtotal + subtotal * tax
    rounded = round(grand)
    return rounded
end

# dendro-expect: duplicate
function final_amount(rows, rate)
    accum = 0
    for row in rows
        accum = accum + row
    end
    whole = accum + accum * rate
    snapped = round(whole)
    return snapped
end

File: test/corpus/rust/duplicates.rs
// An exact clone pair: identical up to renamed identifiers, above the clone size
// floor.

// dendro-expect: duplicate
fn total_price(items: &[i32], tax: i32) -> i32 {
    let mut subtotal = 0;
    for i in 0..items.len() {
        subtotal = subtotal + items[i];
    }
    let grand = subtotal + subtotal * tax;
    let rounded = round_value(grand);
    return rounded;
}

// dendro-expect: duplicate
fn final_amount(rows: &[i32], rate: i32) -> i32 {
    let mut accum = 0;
    for i in 0..rows.len() {
        accum = accum + rows[i];
    }
    let whole = accum + accum * rate;
    let snapped = round_value(whole);
    return snapped;
}
